blog/content/post/2024-03-15-uefi-sb-implementation-mpl.md

```diff
@@ -0,0 +1,361 @@
+`meta-efi-secure-boot`, so we had to remove it:
+
+```bitbake
+# grub-efi_%.bbappend
+
+do_deploy:append:class-target () {
+    # remove default grub-efi-grubx64.efi file deployed by .bb from poky; when
+    # it is left in deploydir, bootimg-efi plugin picks it up after cloning
+    # files from IMAGE_BOOT_FILES list
+    # see: https://git.yoctoproject.org/poky/tree/scripts/lib/wic/plugins/source/bootimg-efi.py?id=00c04394cbc5ecaced7cc1bc8bc8787e621f987d#n360
+    rm -rf ${DEPLOYDIR}/${GRUB_IMAGE_PREFIX}${GRUB_IMAGE}
+}
+```
+
+Now your files should be signed automatically during the build. They will be
+deployed along with their signatures. By default, they will be signed using the
+sample keys from
+[meta-signing-key](https://github.com/Wind-River/meta-secure-core/tree/master/meta-signing-key/files/uefi_sb_keys)
+, which is of course extremely unsafe and should only be used for testing.
+`meta-signing-key` provides a
+[script](https://github.com/Wind-River/meta-secure-core/blob/master/meta-signing-key/scripts/create-user-key-store.sh)
+, which generates custom user keys. Such keys should be enrolled in UEFI so
+that UEFI Secure Boot can be safely utilized.
+
+The script will prompt the user to provide boot key information, such as the
+email address and password. Note that not all generated keys will be used with
+UEFI Secure Boot, as some of them are only compatible with MOK Secure Boot.
+The script finishes by printing lines to be added to your layer's configuration.
+
+```sh
+Enter Boot GPG keyname (use dashes instead of spaces) [default: BOOT-SecureCore]:
+Enter Boot GPG e-mail address [default: SecureCore@foo.com]:
+Enter Boot GPG comment [default: Bootloader Signing Key]:
+Using boot loader gpg name: BOOT-SecureCore
+Using boot loader gpg email: SecureCore@foo.com
+Using boot loader gpg comment: Bootloader Signing Key
+Enter boot loader GPG passphrase: pass
+Enter boot loader locked configuration password(e.g. grub pw): pass
+Creating the user keys for UEFI Secure Boot
+
+(...)
+
+## The following variables need to be entered into your local.conf
+## in order to use the new signing keys:
+
+MASTER_KEYS_DIR = "/path/to/user-keys"
+
+BOOT_KEYS_DIR = "${MASTER_KEYS_DIR}/boot_keys"
+MOK_SB_KEYS_DIR = "${MASTER_KEYS_DIR}/mok_sb_keys"
+SYSTEM_TRUSTED_KEYS_DIR = "${MASTER_KEYS_DIR}/system_trusted_keys"
+SECONDARY_TRUSTED_KEYS_DIR = "${MASTER_KEYS_DIR}/secondary_trusted_keys"
+MODSIGN_KEYS_DIR = "${MASTER_KEYS_DIR}/modsign_keys"
+UEFI_SB_KEYS_DIR = "${MASTER_KEYS_DIR}/uefi_sb_keys"
+GRUB_PUB_KEY = "${MASTER_KEYS_DIR}/boot_keys/boot_pub_key"
+GRUB_PW_FILE = "${MASTER_KEYS_DIR}/boot_keys/boot_cfg_pw"
+
+BOOT_GPG_NAME = "BOOT-SecureCore"
+BOOT_GPG_PASSPHRASE = "pass"
+SIGNING_MODEL = "user"
+
+## Please save the values above to your local.conf
+## Or copy and uncomment the following line:
+# require /path/to/user-keys/keys.conf
+```
+
+Follow the instructions above to use the generated keys in your build.
+
```

Comment:
@PLangowski maybe we could add some demo before summary?

Reply:
We could show the flow of creating and enrolling custom certs and booting signed OS, while other (e.g. Ubuntu) could not boot as the certs were changed

Reply:
Maybe in the form of screenshots from QEMU? What do you think?

Reply:
@PLangowski we have MPL with serial console available. Why not use that and record terminal session with an asciinema?

Reply:
Done 243563d6351104f443c9a52570db65089efb8203

Reply:
Done https://github.com/3mdeb/news-and-ideas/pull/590/commits/243563d6351104f443c9a52570db65089efb8203#diff-408ae19700697ee529e7794c3865363586a3ffe06152e4c90cb578ba42221b40R415